LibSerialize.lua

```diff
@@ -1358,8 +1552,39 @@ function LibSerialize:DeserializeValue(input)
     return unpack(output, 1, outputSize)
 end
 
+function LibSerialize:DeserializeValue(input, opts)
+    local deser = CreateDeserializer(input, opts or {})
+    return deserializeOperation(deser)
+end
+
+function LibSerialize:DeserializeAsyncValue(input, opts)
+    if opts.yieldOnElapsedTime and not opts.timeFn then
+        error("Async Mode operation with yieldOnElapsedTime requires timeFn option")
+    end
+
+    local deser = CreateDeserializer(input, opts or {}, true)
+
+    local thread = coroutine_create(deserializeOperation)
+    -- return coroutine handler
+    return function()
+        local values = {coroutine_resume(thread, deser)}
+        if not values[1] then
+            error(values[2])
+        elseif coroutine_status(thread) ~= 'dead' then
+            return true
+        else
+            return false, unpack(values)
```

Comment:
You're unpacking/returning `values[1]`, which is `co_success`. You can also restructure this to avoid the table intermediary, by using a function intermediary instead. Something like this:

```lua
function checkCoroutineResult(thread, co_success, ...)
    if not co_success then
        return true, false, select(1, ...)
    elseif coroutine_status(thread) ~= "dead" then
        return false
    else
        return true, true, ...
    end
end

local thread = coroutine_create(deserializeOperation)
return function()
    return checkCoroutineResult(thread, coroutine_resume(thread, deser))
end
```

The helper conforms to my earlier suggestion that the handler returns `completed, succeeded, ...`.